modules/local/phyloseq.nf

```diff
@@ -0,0 +1,63 @@
+process PHYLOSEQ {
+    tag "$prefix"
+    label 'process_low'
+
+    conda "bioconda::bioconductor-phyloseq=1.44.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bioconductor-phyloseq:1.44.0--r43hdfd78af_0' :
+        'quay.io/biocontainers/bioconductor-phyloseq:1.44.0--r43hdfd78af_0' }"
+
+    input:
+    tuple val(prefix), path(tax_tsv)
+    path otu_tsv
+    path sam_tsv
+    path tree
+
+    output:
+    tuple val(prefix), path("*phyloseq.rds"), emit: rds
+    path "versions.yml"                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def sam_tsv = "\"${sam_tsv}\""
+    def otu_tsv = "\"${otu_tsv}\""
+    def tax_tsv = "\"${tax_tsv}\""
+    def tree    = "\"${tree}\""
+    def prefix  = "\"${prefix}\""
+    """
+    #!/usr/bin/env Rscript
+
+    suppressPackageStartupMessages(library(phyloseq))
+
+    otu_df  <- read.table($otu_tsv, sep="\\t", header=TRUE, row.names=1)
+    tax_df  <- read.table($tax_tsv, sep="\\t", header=TRUE, row.names=1)
+    otu_mat <- as.matrix(otu_df)
+    tax_mat <- as.matrix(tax_df)
+
+    OTU     <- otu_table(otu_mat, taxa_are_rows=TRUE)
+    TAX     <- tax_table(tax_mat)
+    phy_obj <- phyloseq(OTU, TAX)
+
+    if (file.exists($sam_tsv)) {
```

Comment:
I am not aware of any example for conditional input that uses `file.exists($file)`, so I am a little skeptic here. It might be working fine locally and on github, but we will need to see whether that works also on other systems. But easy enough to change in case it needs to be modified.

Reply:
That's a good point. I did my testing on a Linux server, but I'm hoping it would work on other systems.